lib/validate.go

```diff
@@ -0,0 +1,30 @@
+package lib
+
+import (
+	"fmt"
+	"regexp"
+)
+
+var (
+	validBasicName = regexp.MustCompile("^[a-z0-9_-]+$")
```

Comment:
Question for reviewer: should we allow uppercase characters (`A-Z`)? We want to ensure that we use the same validation as the `datacenter` field in the agent configuration, so that any possible datacenter name is also accepted here.

Ideally, we'd share the validation code between those locations so they don't get out of sync and make the code harder to troubleshoot/maintain.

Reply:
Thank you Jared. I should be modifying the agent startup code to make use of the new lib/validate to bring it to sync.
I will update this PR as soon as I get the confirmation on allowing uppercase characters into the dc name. Right now it only allows characters from - [a-z0-9_-]

Reply:
If the agent configuration code currently only allows `[a-z0-9_-]`, then let's keep it that way. It's just best to share the validation code between those two places if possible to guarantee they are using the same validation. We don't want them to disagree on what's valid!